src/Sylius/Component/Core/Model/Order.php

```diff
@@ -608,4 +609,84 @@ public function getPromotions()
     {
         return $this->promotions;
     }
+
+    /**
+     * Get total value with adjustments from InventoryUnits
+     *
+     * @return int
+     */
+    public function getTotal()
```

Comment:
I'm starting to have second-thoughts about replacing the simple getter logic with an all-in, must calculate every time solution.

From the context of pre-checkout, this approach is rock solid and great. But once you are passed checkout, if you are regularly retrieving order objects and want to the know their total, you are enforcing a full recalculation on your `getTotal` which is going to be prohibitively inefficient and not at all feasible for any significant report.

This brings me onto another through about knowing the `originalTotal` / `unrefundedTotal` / `checkedoutTotal` of the value the customer paid before anything was refunded. It's a core reporting value to know how much money has flowed through the application excluding refunds.  The only way of doing this currently would be to run a query that has to run at the inventoryUnit level excluding any adjustments which are refunds. Again, this is much harder work than storing some original values at order `completion` time.

Let's discuss today.